lightning/src/ln/peer_handler.rs

```diff
@@ -46,10 +46,10 @@ pub struct MessageHandler<CM: Deref, RM: Deref> where
 		RM::Target: RoutingMessageHandler {
 	/// A message handler which handles messages specific to channels. Usually this is just a
 	/// ChannelManager object.
-	pub chan_handler: CM,
+	pub chan_handler: Option<CM>,
 	/// A message handler which handles messages updating our knowledge of the network channel
 	/// graph. Usually this is just a NetGraphMsgHandlerMonitor object.
-	pub route_handler: RM,
+	pub route_handler: Option<RM>,
```

Comment:
Could all the conditional logic be avoided if a constructor took an `Option` but swapped in default / no-op implementations when given `None`?

Reply:
Hmm, that may also make it easier to work with in terms of avoiding "what type is CM, tho?" rustc annoyances when dealing with None. Will do.